src/Tasks/AssemblyDependency/GenerateBindingRedirects.cs

```diff
@@ -131,11 +133,19 @@ public override bool Execute()
 
             if (writeOutput)
             {
+                Log.LogMessageFromResources(MessageImportance.Low, "GenerateBindingRedirects.CreatingBindingRedirectionFile", OutputAppConfigFile.ItemSpec);
                 using (var stream = FileUtilities.OpenWrite(OutputAppConfigFile.ItemSpec, false))
                 {
                     doc.Save(stream);
                 }
             }
+            else if (outputExists)
+            {
+                // if the file exists and the content is up to date, then touch the output file.
```

Comment:
Why? This seems kinda counter to the PR purpose.

Reply:
This is not related to /question flag, but it is a fix the incremental issue.  I call it double incremental issue.
This task is used in a <Target> that has inputs and output.  When the inputs changed, the target will run, but the task would decide to skip (not touching the output).  This would cause the target to run again as its output is not updated.  
Both the task and target are incremental in a slightly different way. This change will force the task to touch the output without modifying the content.